packaging/snapd.mk

```diff
@@ -0,0 +1,190 @@
+
+# Ensure /usr/bin/snapctl is a symlink to /usr/lib/snapd/snapctl
+install:: | $(DESTDIR)$(bindir)
+	ln -s $(libexecdir)/snapd/snapctl $|/snapctl
+
+# Generate and install man page for snap command
+install:: snap | $(DESTDIR)$(mandir)/man8
+	./snap help --man > $|/snap.8
+
+# Install the directory structure in /var/lib/snapd
+install::
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/apparmor/profiles
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/apparmor/snap-confine
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/assertions
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/cache
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/cookie
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/desktop/applications
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/device
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/hostfs
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/lib/{gl,gl32,vulkan}
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/mount
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/seccomp/bpf
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/sequence
+	install -d $(DESTDIR)$(localstatedir)/lib/snapd/snaps
+
+# Touch files that are ghosted by the package. Those are _NOT_ installed but
+# this way the package manager knows about them belonging to the package.
+install:: | $(DESTDIR)$(localstatedir)/lib/snapd
+	touch $|/state.json
+	touch $|/system-key
+
+install:: | $(DESTDIR)$(localstatedir)/cache/snapd
+	touch $|/sections
+	touch $|/names
+	touch $|/commands
+
+install:: | $(DESTDIR)$(snap_mount_dir)
+	touch $|/README
+
+
+# Install the /snap/bin directory
+install::
+	install -d $(DESTDIR)$(snap_mount_dir)/bin
+
+# Install misc directories: 
+install::
+	install -d $(DESTDIR)$(localstatedir)/cache/snapd
+	install -d $(DESTDIR)$(datadir)/polkit-1/actions
+
+# Remove traces of ubuntu-core-launcher. It is a phased-out executable that is
+# still partially present in the tree but should be removed in the subsequent
+# release.
+install::
+	rm -f $(DESTDIR)$(bindir)/ubuntu-core-launcher
+
+ifeq ($(with_core_bits),0)
+# Remove systemd units that are only used on core devices.
+install::
+	rm -f $(addprefix $(DESTDIR)$(unitdir)/,snapd.autoimport.service snapd.system-shutdown.service snapd.snap-repair.timer snapd.snap-repair.service snapd.core-fixup.service)
+
+# Remove fixup script that is only used on core devices.
+install::
+	rm -f $(DESTDIR)$(libexecdir)/snapd/snapd.core-fixup.sh
+
+# Remove system-shutdown helper that is only used on core devices.
+install::
+	rm -f $(DESTDIR)$(libexecdir)/snapd/system-shutdown
+endif
+
+ifeq ($(with_apparmor),0)
+# Don't ship apparmor helper service when AppArmor is not enabled.
+install::
+	rm -f $(DESTDIR)$(unitdir)/snapd.apparmor.service
+	rm -f $(DESTDIR)$(libexecdir)/snapd/snapd-apparmor
+endif
+
+# Tests use C.UTF-8 because some some code depend on this for fancy Unicode
+# output that unit tests do not mock.
+.PHONY: check
+check:
+	LC_ALL=C.UTF-8 go test $(import_path)/...
```

Comment:
No C.UTF-8 on Arch. Maybe it's best not to set anything here.

Reply:
We must, otherwise tests are failing. How come arch has no C.UTF_8 locale?!? It's the best thing libc had introduced in years.

Reply:
> It's the best thing libc had introduced in years.
That's the catch, it's not glibc but downstream distros that introduced it.

Reply:
Apparently they are the sane ones then :)

Reply:
I believe we can tweak this as we get to apply this to Arch.